include/bbp/sonata/reports.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+/**
+ * Provides a bridge between c-based programs in order to have access to the c++-based objects
+ */
+#ifndef __cplusplus
+#include <stddef.h>
+#endif
+#if defined(__cplusplus)
+#include <cstddef>
+#include <stdint.h>
+extern "C" {
+#endif
+
+
+/*! \brief Add node to existing or new report
+ * @param report_name name of the report to be created or added node to
+ * @param gid node/cell identifier
+ * @param tstart start time of recording for the report
+ * @param tend stop time of recording for the report
+ * @param dt frequency of recordings
+ * @param kind type of report (soma, compartment)
+ * @return 0 if operator succeeded, -1 otherwise
+ */
+int sonata_create_report(
+    const char* report_name, double _tstart, double _tend, double _dt, const char* kind);
+
+/*!
+ * \brief Add a node to an existing report
+ * \param report_name name of the report
+ * \param node_id node identifier
+ * \return 0 if operator succeeded, -2 if the report doesn't exist, -1 for other errors.
+ */
+int sonata_add_node(const char* report_name, uint64_t node_id);
```

Comment:
what happen if we add 2 time the same node ? do we have to check for existence first ?

Reply:
It is being checked (returned -1). Will update the docstring.

Reply:
Couldn't we add exceptions here for that case?